maistra/bazelrc

```diff
@@ -5,6 +5,7 @@ build --cxxopt -w
 
 build --copt -DOPENSSL_IS_BORINGSSL=0
 build --verbose_failures
+build -c opt
```

Comment:
I feel `build:ci-config -c opt` would be better; that would make this the default for our ci-environment.. and maybe a one liner explaining why we do this would be helpful too for future maintenance.

Reply:
I agree with Otto on this,  while a RELEASE configuration is ideal we need to identify the problem this change addresses.

Reply:
I see; I am blissfully unaware of the details around that :-) but my first (perhaps naive) thought is that often such "heisenbugs" that only occur when certain compiler flags are enabled, are in fact caused by undefined behaviour.. In which case I'd rather see us run Envoy's tests with ubsan in CI (the fact that `-c opt` teases out such issues occasionally is not guaranteed, but ubsan should reliably point out certain types of undefined behaviour as long as test coverage is decent).

This thought doesn't just apply here, thinking about it, it also applies to proxy. In retrospect, I'm actually curious if running proxy tests with sanitizers enabled would have caught https://istio.io/latest/news/security/istio-security-2022-005/ earlier.